### PR TITLE
User Authorization Handling

### PR DESCRIPTION
**Pull request by chanchana**
Add
- Authorization middleware
- Get user from request. remove all request that rely user data from payload
- Check permission for some endpoint

Frontend:
- Remove `Auth` component
- Remove sending userid in payload to server

### CODE REVIEW

File: api/report.py [hunk 1]
```diff
@@ -45,6 +46,7 @@ def post(self):
         campaign.save()
         return new_report.dict()
 
+@DeprecationWarning
```
Comment (justinasr) [inline api/report.py:49]: Why is this here?

File: api/report.py [hunk 2]
```diff
@@ -94,12 +96,21 @@ class ReportAPI(Resource):
...
+        if report.group not in user.groups:
+            raise Forbidden()
+        if 'authors' in api.payload:
+            api.payload.pop('authors')
         report.update(api.payload)
+        if 'content' in api.payload:
+            previous_authors = []
+            if report.authors:
+                previous_authors = [author for author in report.authors if author.id != user.id]
+            new_authors = [user] + previous_authors
```
Comment (justinasr) [inline api/report.py:113]: What is the point of first filtering out the user from the authors and then adding it back?

File: api/report.py (hunk 1 above)
Comment (chanchana) [reply api/report.py:49]: I just mark that this class will be removed since we don't need my report page anymore.

File: api/report.py (hunk 2 above)
Comment (chanchana) [reply api/report.py:113]: - To make the most recent author who edit the report be the fist in the list.
- To prevent duplicate users if they edit the report more than one time.
